tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,739 @@
+# TODO(mabdinur): Move this file to tests/test_the_test/
```

Comment:
TODO to be addressed :)

Reply:
I'm actually wondering if really we want to have this in test the test folder ?

Reply:
I will remove this comment. This PR is already doing a lot. We can discuss the merits of this change in a future PR.